examples/platforms/simulation/radio.c

```diff
@@ -221,6 +221,42 @@ static uint16_t crc16_citt(uint16_t aFcs, uint8_t aByte)
     return (aFcs >> 8) ^ sFcsTable[(aFcs ^ aByte) & 0xff];
 }
 
+
+#if OPENTHREAD_SIMULATION_VIRTUAL_TIME == 1
+void reportRadioStatusToOTNS(void)
+{
+    struct Event event;
+
+    event.mDelay      = 1; // 1us for now
+    event.mEvent      = OT_SIM_EVENT_OTNS_STATUS_PUSH;
+
+    switch (sState)
+    {
+    case OT_RADIO_STATE_RECEIVE:
+        event.mDataLength = 8;
+        memcpy(event.mData, "radio=rx", 8);
```

Comment:
To be honest, Initially I designed this report system using a whole new event (instead of Status push). This event would just transmit a single byte with the respective radio status code. The reason is that these events of radio transition will happen too often. So, for speed and to avoid delaying radio tasks, I wanted to optimize this procedure the best could (perhaps these reports could be used in real devices in the future).

However, I decide to keep the event inside "Status Push" to keep the standard format already used in this platform. But still, I question: the use of Sprintf and slower functions is a good future prove idea? I bet it can't affect too much the computers nowadays, but what do you think?

Reply:
It should have little to no impact on the efficiency of OpenThread simulations. 
Currently, the major bottleneck of OTNS running with OT simulation is the delay caused by UDP packet transmissions.

Reply:
After a second read of your suggestion, I think I got the main goal of it. So, considering that the impact is minimal, what do you think of this:
```
void reportRadioStatusToOtns(void)
{
    struct Event event;

    event.mDelay      = 1; // 1us for now
    event.mEvent      = OT_SIM_EVENT_OTNS_STATUS_PUSH;

    char const *stateString;
    switch (sState)
    {
    case OT_RADIO_STATE_RECEIVE:
        stateString = "rx";
        break;
    case OT_RADIO_STATE_TRANSMIT:
        stateString = "tx";
        break;
    case OT_RADIO_STATE_DISABLED:
        stateString = "off";
        break;
    case OT_RADIO_STATE_SLEEP:
        stateString = "sleep";
        break;
    default:
        fprintf(stderr, "Invalid radio status: %d\n", sState);
        return;
    }

    int n = snprintf((char *)event.mData, sizeof(event.mData), "radio_state=%s", stateString);
    if (n < 0)
    {
        fprintf(stderr, "Failed to format radio status: %d\n", n);
        return;
    }

    event.mDataLength = n;

    otSimSendEvent(&event);
}
```